tools/configure_server_settings.rb

```diff
@@ -32,7 +39,7 @@
 keys.each { |p| path = path[p.to_sym] }
 
 puts "Setting [#{opts[:path]}], old value: [#{path[key]}], new value: [#{opts[:value]}]"
```

Comment:
Should we cherry-pick the first commit from #17043 so we can see the class?

Or perhaps we don't need to care about telling the user about the string/integer class unless they don't match:
```ruby
# allow user to escape if the new value's class is not the same as the original,
# such as setting a String where it was previously an Integer
if path[key] && path[key].class != opts[:value].class
  STDERR.puts "The new value's class #{opts[:value].class} does not match the prior one's #{path[key].class}.  Do you want to continue? (Y/N)"
  exit 1 if STDERR.gets.strip.downcase == "n"
end  


```

Reply:
Perhaps the above condition could advise the user to run `tools/configure_server_settings.rb -h` to see how to specify Integer.

Reply:
That is a good idea, essentially being able to tell the user that the input made is not valid.